File: tests/xcalablemp/NOT-IMPLEMENTED/23/array9.f90
! testp089.f
! loop指示文とarray指示文のテスト

      program main
      include 'xmp_lib.h'
      integer,parameter:: N=1000
!$xmp nodes p(*)
!$xmp template t1(N,N,N)
!$xmp template t2(N,N,N)
!$xmp template t3(N,N,N)
!$xmp distribute t1(*,*,gblock((/325,435,111,129/))) onto p
!$xmp distribute t2(*,gblock((/325,435,111,129/)),*) onto p
!$xmp distribute t3(gblock((/325,435,111,129/)),*,*) onto p
      integer a(N)
      real*8  b(N)
      real*4  c(N)
!$xmp distribute a(i) with t1(*,*,i)
!$xmp distribute b(i) with t2(*,i,*)
!$xmp distribute c(i) with t3(i,*,*)
      character(len=2) result

      result = 'OK'
!$xmp loop on t1(:,:,i)
      do i=1, N
!$xmp array on t1(:,:,i)
         a(i) = i
      enddo
!$xmp loop on t2(:,i,:)
      do i=1, N
!$xmp array on t2(:,i,:)
         b(i) = dble(i)
      enddo
!$xmp loop on t3(i,:,:)
      do i=1, N
!$xmp array on t3(i,:,:)
         c(i) = real(i)
      enddo

!$xmp loop on t1(:,:,i)
      do i=1, N
         if(a(i) .ne. i) result = 'NG'
      enddo
!$xmp loop on t2(:,i,:)
      do i=1, N
         if(b(i) .ne. dble(i)) result = 'NG'
      enddo
!$xmp loop on t3(i,:,:)
      do i=1, N
         if(c(i) .ne. real(i)) result = 'NG'
      enddo

      print *, xmp_node_num(), 'testp089.f ', result

      end
      

File: tests/xcalablemp/NOT-IMPLEMENTED/23/array9.c
/*testp089*/
/*loop指示文とarray指示文のテスト*/
#include<xmp.h>
#include<mpi.h>
#include<stdio.h>      
#pragma xmp nodes p(*)
static const int N=1000;
#pragma xmp template t1(0:N-1,0:N-1,0:N-1)
#pragma xmp template t2(0:N-1,0:N-1,0:N-1)
#pragma xmp template t3(0:N-1,0:N-1,0:N-1)
#pragma xmp distribute t1(*,*,gblock((/325,435,111,129/))) onto p
#pragma xmp distribute t2(*,gblock((/325,435,111,129/)),*) onto p
#pragma xmp distribute t3(gblock((/325,435,111,129/)),*,*) onto p
int a[N];
double b[N];
float c[N];
#pragma xmp align a[i] with t1(*,*,i)
#pragma xmp align b[i] with t2(*,i,*)
#pragma xmp align c[i] with t3(i,*,*)
int i;
char *result;
int main(void){
  
   result = "OK";
#pragma xmp loop on t1(:,:,i)
   for(i=0;i<N;i++){
#pragma xmp array on t1(:,:,i)
      a[i] = i;
   }
#pragma xmp loop on t2(:,:,i)
   for(i=0;i<N;i++){
#pragma xmp array on t2(:,:,i)
      b[i] = (double)i;
   }
#pragma xmp loop on t3(:,:,i)
   for(i=0;i<N;i++){
#pragma xmp array on t3(:,:,i)
      c[i] = (float)i;
   }

#pragma xmp loop on t1(:,:,i)
   for(i=0;i<N;i++){
      if(a[i] != i) result = "NG";
   }
#pragma xmp loop on t2(:,:,i)
   for(i=0;i<N;i++){
      if(b[i] != (double)i) result = "NG";
   }
#pragma xmp loop on t3(:,:,i)
   for(i=0;i<N;i++){
      if(a[i] != (float)i) result = "NG";
   }
   
   printf("%d %s %s\n",xmp_node_num(),"testp089.c",result);
   return 0;
}    
         
      
   
